interview-questions.md

```diff
@@ -4,55 +4,64 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A function is a block of reusable code that takes in an argument or arguments and outputs a result.
 
-  Researched answer:
+  Researched answer: "A function is a JavaScript procedure—a set of statements that performs a task or calculates a value."
+  Source - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions
 
 
 
 2. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: The difference between map and filter is that map loops through an array and unselectively creates an additional array while filter loops through an array with an if conditional creating a selective "filter".
 
-  Researched answer:
+  Researched answer: "They both return a new array. map returns a new array of elements where you have applied some function on the element so that it changes the original element (typically). filter returns a new array of the elements of the original array (with no change to the original values). filter will only return elements where the function you specify returns a value of true for each element passed to the function."
+  Source - https://forum.freecodecamp.org/t/the-difference-between-map-and-filter/277659
 
 
 
 3. What is the difference between map/filter and a for loop?
 
-  Your answer:
+  Your answer: Map/filter are higher order functions that create a new array from the original. For loop only runs through the original array without creating another array.
 
-  Researched answer:
+  Researched answer: "Loops are generally used, in any programming language, to perform operations on arrays: given an array you can iterate over its elements and perform a calculation."
+  "map returns an array with the same length,"
+  "filter as the name implies, it returns an array with less"
+  Source - https://flaviocopes.com/javascript-loops-map-filter-reduce-find/
 
 
 
 4. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: The difference is that return is used as a mechanism in a function to output a value and console.log places a value into a function which is used for development testing purposes.
```

Comment:
Just about! You can console.log ANYTHING and it doesn't have to strictly stay inside of a function. Still a pretty good answer!